dtrenz/LaunchGate
# Updating to JSONDecoder from JSONSerialization

## What you did :question:
- Updated `parse` method: uses JSONDecoder instead of JSONSerialization

## How you did it :white_check_mark:
- Made model classes (LaunchGateConfiguration, UpdateConfiguration, and AlertConfiguration) conform to Decodable
- Manually decoded the LaunchGateConfiguration and UpdateConfiguration classes with decoder initializers
- Rewrote `parse` method to use JSONDecoder in order to parse the root object (LaunchGateConfiguration type) and its nested objects (UpdateConfiguration and AlertConfiguration)
- Removed error enum, since Decodable classes will throw the appropriate errors

## How to test it :microscope:
- Build and run the Example app
- An alert should pop up that reads "An update is required to continue using this app."


## Screenshots (if applicable) :camera:
![simulator screen shot - iphone xr - 2018-11-16 at 09 35 01](https://user-images.githubusercontent.com/8409475/48627665-237a3800-e983-11e8-8c65-f2ca091b0a50.png)
